crates/node/src/tee/remote_attestation.rs

```diff
@@ -85,25 +91,80 @@ pub async fn submit_remote_attestation(
         .context("failed to submit attestation after multiple retry attempts")?
 }
 
+fn validate_remote_attestation(
+    attestation: &Attestation,
+    tls_public_key: Ed25519PublicKey,
+    account_public_key: Ed25519PublicKey,
+    allowed_docker_image_hashes: &[MpcDockerImageHash],
+    allowed_launcher_compose_hashes: &[LauncherDockerComposeHash],
+) -> Result<(), VerificationError> {
+    let expected_report_data =
+        ReportData::new(*tls_public_key.as_bytes(), *account_public_key.as_bytes());
+    let now = std::time::SystemTime::now()
+        .duration_since(std::time::UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
+    attestation.verify(
+        expected_report_data,
+        now,
+        allowed_docker_image_hashes,
+        allowed_launcher_compose_hashes,
+    )
+}
+
+pub async fn validate_and_submit_remote_attestation(
+    tx_sender: impl TransactionSender,
+    attestation: Attestation,
+    tls_public_key: Ed25519PublicKey,
+    account_public_key: Ed25519PublicKey,
+    allowed_image_hashes_in_contract: watch::Receiver<Vec<MpcDockerImageHash>>,
+    allowed_launcher_compose_hashes_in_contract: watch::Receiver<Vec<LauncherDockerComposeHash>>,
```

Comment:
Nit: I don't see why this function needs to take a `warch::Receiver` as opposed to just receiving a reference to the latest values

Reply:
I needed to select the proper place to convert from `watch::Receiver` to the reference to the latest values. The place I chose was basically `validate_and_submit_remote_attestation` function. What place do you suggest for doing that?